libcxx/include/__flat_map/container_traits.h

```diff
@@ -0,0 +1,46 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#ifndef _LIBCPP___FLAT_MAP_CONTAINER_TRAITS_H
+#define _LIBCPP___FLAT_MAP_CONTAINER_TRAITS_H
+
+#include <__config>
+
+#if !defined(_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER)
+#  pragma GCC system_header
+#endif
+
+#if _LIBCPP_STD_VER >= 23
+
+_LIBCPP_BEGIN_NAMESPACE_STD
+
+template <class _Tp>
+inline constexpr bool __is_stl_container = false;
+
+template <class _Tp>
+struct __container_traits {
+  static constexpr bool __emplacement_has_strong_exception_safety_guarantee = false;
+};
```

Comment:
Should we just use a variable template?

Reply:
originally i was planning to add `__clear_has_strong_exception_safety_guarantee`. but in the end, I don't need it.  However, I feel that it is better to keep a `traits` class as a holder of all the `traits` we need

Reply:
This part is taken out to https://github.com/llvm/llvm-project/pull/109578